src/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,187 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+
+/**
+ * The XSLTInjection plugin identifies XSLT injection vulnerabilities with all the parameters in GET and POST
+ *
+ * @author CaptainFreak
+ */
+public class XSLTInjection extends AbstractAppParamPlugin {
+
+	// wasc_23 is XML Injection
+	private static Vulnerability vuln = Vulnerabilities.getVulnerability("wasc_23");
+    
+	// plugin dependencies
+    private static final String[] dependency = {};
+    
+	//For logging
+	private static Logger log = Logger.getLogger(XSLTInjection.class);
+    private boolean debugEnabled = log.isDebugEnabled();
+
+    // Payloads
+    
+    private static final String[] errorCausingPayloads={"<",">","\'","\""};
+    
+    private static final String[] vendorReturningPayloads={"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/>",
+    														"system-property(\'xsl:vendor\')/>",
+    														"\"/><xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--",
+    														"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--"
+    														};
+    private static final String[] xsltVendors={"libxslt","Microsoft","Saxonica","Apache","Xalan","SAXON","Transformiix"};
+    
+    //TODO Add time based payloads (SSRF)
+    
+	@Override
+	public int getId() {
+		return 90017;
+	}
+
+	@Override
+	public String getName() {
+		
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return dependency;
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {  
+		return technologies.includes(Tech.XML); 
+	}
+	
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.refs");
+	}
+
+	@Override
+	public void init() {
+        //DEBUG: turn on for debugging
+        //log.setLevel(org.apache.log4j.Level.DEBUG);
+        //this.debugEnabled = true;
```

Comment:
If `setLevel` is `DEBUG` then `isDebugEnabled()` returns true, no need to set two things...